PowerPointLabs/PowerPointLabs/ZoomLab/AutoZoom.cs

```diff
@@ -460,5 +454,31 @@ private static void SelectAllShape(List<PowerPoint.Shape> shapes)
                 shapeCopy.Select(Office.MsoTriState.msoFalse);
             }
         }
+
+        private static PowerPoint.Shape AddSlideAsShape(PowerPointSlide slideToAdd, PowerPointSlide targetSlide)
+        {
+            try
+            {
+                // Export the slide as .png to a temporary location, then add it to shapes.
+                // This yields a higher quality image compared to copy-pasting slide as image.
+                string tempFilePath = FileDir.GetTemporaryPngFilePath();
+                GraphicsUtil.ExportSlide(slideToAdd, tempFilePath);
+                PowerPoint.Shape slideAsShape = targetSlide.Shapes.AddPicture2(tempFilePath,
+                                                                                 Microsoft.Office.Core.MsoTriState.msoFalse,
+                                                                                 Microsoft.Office.Core.MsoTriState.msoTrue,
+                                                                                 0,
+                                                                                 0);
+                FileDir.DeleteFile(tempFilePath);
```

Comment:
Should FileDir.DeleteFile have its own try catch?
I'm worried that if everything succeeds except delete file, then there will be a duplicate of the shape on the slide. This is because the adding of the picture to the slide has already succeeded, then deletefile throws an exception but yet the code will go to the "catch" block and then run the copy and paste code. This results in two shapes on the slide?

Reply:
Hmm, I was operating under the assumption that if the user was able to create a file, user should also be able to delete that same file. But I guess it wouldn't hurt to be a little safer. Made the changes as suggested!